unittests/AST/ASTImporterTest.cpp

```diff
@@ -294,24 +325,15 @@ class ASTImporterTestBase : public ::testing::TestWithParam<ArgVector> {
   // May be called several times in a given test.
   // The different instances of the param From may have different ASTContext.
   Decl *Import(Decl *From, Language ToLang) {
-    if (!ToAST) {
-      ArgVector ToArgs = getArgVectorForLanguage(ToLang);
-      // Build the AST from an empty file.
-      ToAST =
-          tooling::buildASTFromCodeWithArgs(/*Code=*/"", ToArgs, "empty.cc");
-      ToAST->beginSourceFile();
-    }
-
-    // Create a virtual file in the To Ctx which corresponds to the file from
-    // which we want to import the `From` Decl. Without this source locations
-    // will be invalid in the ToCtx.
-    auto It = std::find_if(FromTUs.begin(), FromTUs.end(), [From](const TU &E) {
-      return E.TUDecl == From->getTranslationUnitDecl();
-    });
-    assert(It != FromTUs.end());
-    createVirtualFileIfNeeded(ToAST.get(), It->FileName, It->Code);
+    lazyInitToAST(ToLang);
+    TU *FromTU = findFromTU(From);
+    return FromTU->import(ToAST.get(), From);
+  }
 
-    return It->import(ToAST.get(), From);
+  QualType ImportType(QualType FromType, Decl *FromDecl, Language ToLang) {
```

Comment:
Why do we need the decl? Only to get the ASTContext? If that is the case maybe it would be better to ask for an AST Context in the API?

Reply:
TU objects are found in FromTUs vector by TUDecl equivalence in order to use the correct importer. This is happening because our API, not theirs.

Reply:
Maybe we could ask for a TU decl then? The reason is that I find the current API very confusing. FromType and FromDecl suggest that there is a correlation between the type and the decl other than being in the same TU.

Reply:
So you are saying renaming FromDecl to TUDecl would help the readability?

Reply:
Yeah.

Reply:
Thanks, done.